notification/search.go

```diff
@@ -67,6 +67,15 @@ var searchTemplate = template.Must(template.New("search").Funcs(search.Helpers()
 	SELECT
 		(trunc((extract('epoch' from om.created_at)-:timeSeriesOrigin)/:timeSeriesInterval))::bigint AS bucket,
 		count(*)
+		{{if .SegmentByService}}
+		, s.name
+		{{end}}
+		{{if .SegmentByUser}}
+		, u.name
+		{{end}}
+		{{if .SegmentByMessageType}}
+		, om.message_type
+		{{end}}
```

Comment:
I expect them to be functionally equiv. but is there a reason this is using different logic than the GROUP BY below? (e.g., `else if and .TimeSeries .SegmentByUser`) if they were identical it would be more clear they work the same

Reply:
I did the else if and was getting errors on my syntax, I'll mess with it more to try that approach 👍

Reply:
Though I have it nested inside of the first `{{.if .TimeSeries}}` so we don't have to rewrite out
```
SELECT (trunc((extract('epoch' from om.created_at)-:timeSeriesOrigin)/:timeSeriesInterval))::bigint AS bucket,
```
several times, since in the group by doing `GROUP BY bucket` a few times isn't as noticeable

Reply:
So maybe for group by we can just do:
```sql
	{{if .SegmentByService}}
	GROUP BY bucket, s.name
	{{else if .SegmentByUser}}
	GROUP BY bucket, u.name
	{{else if .SegmentByMessageType}}
	GROUP BY bucket, om.message_type
```

As for the `else if` -- ping me the error and I can try to help; but it would make it clear that only one (at most) of the three will end up in the query

Reply:
That is how the group by is currently, so we should be good 👍